Doc/c-api/function.rst

```diff
@@ -118,3 +118,56 @@ There are a few functions specific to Python functions.
    must be a dictionary or ``Py_None``.
 
    Raises :exc:`SystemError` and returns ``-1`` on failure.
+
+
+.. c:function:: int PyFunction_AddWatcher(PyFunction_WatchCallback callback)
+
+   Register *callback* as a function watcher for the current interpreter. Returns
+   an id which may be passed to :c:func:`PyFunction_ClearWatcher`. In case
+   of error (e.g. no more watcher IDs available), return ``-1`` and set an
+   exception.
```

Comment:
Nit: semantic line-breaks will make this paragraph more neatly formatted. Personal preference only; you may ignore this suggestion. Also, consistently use imperative mood.

```suggestion
   Register *callback* as a function watcher for the current interpreter.
   Return an id which may be passed to :c:func:`PyFunction_ClearWatcher`.
   In case of error (e.g. no more watcher IDs available),
   return ``-1`` and set an exception.
```

Reply:
Also, throughout the PR you use both lower-case `id` and upper-case `ID` (docs, comments, exception messages, etc.). I'd consider sticking to one of them for consistency.